C/services/common-plugin-interfaces/python/include/python_plugin_common_interface.h

```diff
@@ -0,0 +1,479 @@
+#ifndef _PYTHON_PLUGIN_BASE_INTERFACE_H
+#define _PYTHON_PLUGIN_BASE_INTERFACE_H
+/*
+ * FogLAMP common plugin interface
+ *
+ * Copyright (c) 2019 Dianomic Systems
+ *
+ * Released under the Apache 2.0 Licence
+ *
+ * Author: Massimiliano Pinto, Amandeep Singh Arora
+ */
+
+using namespace std;
+
+extern "C" {
+// This is the Python object initialised in:
+// South, Notification, Filter and North python plugin interfaces
+extern PyObject* pModule;
+
+// PluginName set by PluginInterfaceInit
+extern string sPluginName;
+
+// Common methods to all plugin interfaces
+static PLUGIN_INFORMATION *plugin_info_fn();
+static PLUGIN_HANDLE plugin_init_fn(ConfigCategory *);
+static void plugin_reconfigure_fn(PLUGIN_HANDLE*, const std::string&);
+static void plugin_shutdown_fn(PLUGIN_HANDLE);
+
+static void logErrorMessage();
+
+/**
+ * Destructor for PythonPluginHandle
+ *    - Free up owned references
+ *    - Unload python 3.5 interpreter
+ */
+static void PluginInterfaceCleanup()
```

Comment:
PluginInterfaceInit is in south-plugin-interfaces/python/python_plugin_interface.cpp 
and it will be in

notification--plugin-interfaces/python/python_plugin_interface.cpp 

etc

This PluginInterfaceInit can be slightly different for each interface.